ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -546,6 +578,57 @@ public class ResourceDownloadManager {
     }
   }
 
+  public func standardKeyboardInstallCompletionBlock(forFullID fullID: FullKeyboardID, withModel: Bool = true) -> CompletionHandler<InstallableKeyboard> {
+    return { package, error in
+      if let package = package {
+        // The "keyboard package" is actually a cloud resource that was already directly installed.
+        // So, we don't 'install it from the package'; we just note that it's already present.
+        let fullID = FullKeyboardID(keyboardID: fullID.keyboardID, languageID: fullID.languageID)
+
+        // Since we're installing from a cloud resource, the files are already appropriately placed.
+        // We just need to actually add the resource.
+        if let keyboard = package.findResource(withID: fullID) {
+          ResourceFileManager.shared.addResource(keyboard)
+          _ = Manager.shared.setKeyboard(keyboard)
+
+          if withModel {
+            // TODO:  Adapt as appropriate.
+            self.downloadLexicalModelsForLanguageIfExists(languageID: fullID.languageID)
+          }
+        } else {
+          log.error("Expected resource \(fullID.description) download failed; resource unexpectedly missing")
+        }
+      } else if let error = error {
+        log.error("Installation failed: \(String(describing: error))")
+      }
+    }
+  }
+
+  public func standardLexicalModelInstallCompletionBlock(forFullID fullID: FullLexicalModelID) -> CompletionHandler<InstallableLexicalModel> {
+    return { package, error in
+      if let package = package {
+        do {
+          // A raw port of the queue's old installation method for lexical models.
+          try ResourceFileManager.shared.finalizePackageInstall(package, isCustom: false)
```

Comment:
Note - at this stage, we could _very_ easily just install the language pairing that was requested.

That said, I believe we've stated an intention to install lexical models for all supported languages, rather than just the one.  Either way, the plan is to (eventually) replace the `finalizePackageInstall` call with something more explicit.

Currently kept in this form for ease of comparison to [lines 525-535 of ResourceDownloadQueue](https://github.com/keymanapp/keyman/pull/3277/files#diff-ea275df5a28f3b6b121ac8f8c846fc58L525).